Docs updates

dwhswenson opened the pull request:
Some cleanup for docs and some new materials added.

* Removed the "Guides" heading (as separate from "Topics") and moved everything into "Topics". It wasn't clear to me what the distinction was supposed to be, and I'm pretty sure it wouldn't be clear to users. I also don't show a couple things that I don't think were really meant to be for users (like levels of user experience -- that was supposed to be internal for discussion purposes)
* New docs for data objects (some of which taken from the current draft of the paper)
* Lots of little cleanup to fix warnings that sphinx was throwing.
* Removed the "Full API" link, which wasn't really full, and was raising warnings in Sphinx. Maybe we should look at `sphinx-apidoc` and see if we can get actual complete API documentation from that?
* Automated the docs versioning using `pkg_resources` and `packaging.version`. Requires installing OPS before building docs, but I don't think that's a problem (needs to be installed to build anyway, doesn't it?) Makes one less place to update version numbers.

dwhswenson commented:
Passes test. Ready for review.

dwhswenson commented:
Could this get a review ASAP? I'll be telling people about OPS at the "Recent Advances in Modeling Rare Events" conference at the end of this week, and I'd rather point them to the website *after* these changes have been made.

jhprinz commented on docs/topics/absolute_beginners.rst:
```diff
@@ -0,0 +1,103 @@
+Collective variables are functions of a snapshot
+================================================
+
+For a system with :math:`N` atoms, each snapshot on a trajectory (in 3D
```
We might explain the use of _snapshot_ as a frozen picture of the system state at a particular in time (which is more or can be more than just a phase space point, etc...)

jhprinz commented on docs/topics/absolute_beginners.rst:
```diff
@@ -0,0 +1,103 @@
+We also have a class of objects called :class:`.Ensemble`. More correctly,
+we should call these "path ensembles" (or even more correctly, "path
+ensemble functions.") The most important function this provides is the
+indicator function, which tells you whether or a given trajectory is in the
```
remove _or_

jhprinz commented on docs/topics/ensembles_apply_to_trajectories_not_frames.rst:
```diff
@@ -0,0 +1,97 @@
+==================================================
+
+
+In might seem like ``Ensemble(volume_A) & Ensemble(volume_B)`` should be the
```
It ...

dwhswenson commented:
Fixed those typos. Also removed the wrapping of `.md` files in `.rst`, done in `premake.py`, since I converted the Markdown to `.rst`. Going to merge this now, since the changes were approved!
